### PR TITLE
[flink] Introduce PartitionMarkDone action and procedure

### PR DESCRIPTION
**Pull request by wwj6591812**
### Purpose

This pr is introduce PartitionMarkDone action and procedure for user can PartitionMarkDone at any time.

Linked issue: close #xxx

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    @Override
+    public void run() throws Exception {
+        List<org.apache.paimon.flink.sink.partition.PartitionMarkDoneAction> actions =
+                getPartitionMarkDoneActions();
+        for (LinkedHashMap<String, String> partition : partitions) {
+            try {
+                for (org.apache.paimon.flink.sink.partition.PartitionMarkDoneAction action :
+                        actions) {
+                    action.markDone(PartitionPathUtils.generatePartitionPath(partition));
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java:77]: Can we introduce a `PartitionPathUtils.generatePartitionPath(Map<String, String> partitionSpec, RowType partitionType)`? `LinkedHashMap` can be generated in it.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                                            checkMetastoreAndCreateMetastoreClient(
+                                                    fileStoreTable, coreOptions, options));
+                                default:
+                                    throw new UnsupportedOperationException(action);
+                            }
+                        })
+                .collect(Collectors.toList());
+    }
+
+    private MetastoreClient checkMetastoreAndCreateMetastoreClient(
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java:105]: just name it `createMetastoreClient`

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        return new String[] {"Success"};
+    }
+
+    private List<PartitionMarkDoneAction> getPartitionMarkDoneActions(
+            FileStoreTable fileStoreTable, CoreOptions coreOptions, Options options) {
+        return Arrays.asList(options.get(PARTITION_MARK_DONE_ACTION).split(",")).stream()
+                .map(
+                        action -> {
+                            switch (action) {
+                                case "success-file":
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java:98]: Extract a static method in `PartitionMarkDoneAction`.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        Table table = catalog.getTable(identifier);
+        checkArgument(
+                table instanceof FileStoreTable,
+                "Only FileStoreTable supports partition_mark_done procedure. The table type is '%s'.",
+                table.getClass().getName());
+
+        FileStoreTable fileStoreTable = (FileStoreTable) table;
+        CoreOptions coreOptions = fileStoreTable.coreOptions();
+        Options options = coreOptions.toConfiguration();
+        List<org.apache.paimon.flink.sink.partition.PartitionMarkDoneAction> actions =
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java:74]: You should close actions.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java (hunk 1 above)
Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java (hunk 2 above)
Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/PartitionMarkDoneAction.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java (hunk 4 above)
Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java:69]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java (hunk 3 above)
Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PartitionMarkDoneProcedure.java]: done

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**